pennylane/transforms/get_dag_commutation.py

```diff
@@ -0,0 +1,589 @@
+
+        # Case 3.1: control and target 1 overlap with target 2
+        if control_target and target_target:
+            return bool(
+                commutation_map[operation1.is_controlled][position[operation2.name]]
+            ) and bool(commutation_map["ctrl"][position[operation2.name]])
+
+        # Case 3.2: control operation 1 overlap with target 2
+        if control_target and not target_target:
+            return bool(commutation_map["ctrl"][position[operation2.name]])
+
+        # Case 3.3: target 1 overlaps with target 2
+        if not control_target and target_target:
+            return bool(commutation_map[operation1.is_controlled][position[operation2.name]])
+
+    # Case 4: only operation 2 is controlled
+    elif operation2.is_controlled:
+        target_control = intersection(operation1.wires, operation2.control_wires)
+        target_target = intersection(operation1.wires, operation2.target_wires)
+
+        # Case 4.1: control and target 2 overlap with target 1
+        if target_control and target_target:
+            return bool(
+                commutation_map[operation1.name][position[operation2.is_controlled]]
+            ) and bool(commutation_map[operation1.name][position[operation2.is_controlled]])
+
+        # Case 4.2: control operation 2 overlap with target 1
+        if target_control and not target_target:
+            return bool(commutation_map[operation1.name][position["ctrl"]])
+
+        # Case 4.3: target 1 overlaps with target 2
+        if not target_control and target_target:
+            return bool(commutation_map[operation1.name][position[operation2.is_controlled]])
+
+    # Case 5: no controlled operations
+    # Case 5.1: no controlled operations we simply check the commutation table
+    return bool(commutation_map[operation1.name][position[operation2.name]])
+
+
+def _merge_no_duplicates(*iterables):
+    """Merge K list without duplicate using python heapq ordered merging
+
+    Args:
+        *iterables: A list of k sorted lists
+
+    Yields:
+        Iterator: List from the merging of the k ones (without duplicates
+    """
+    last = object()
+    for val in heapq.merge(*iterables):
+        if val != last:
+            last = val
+            yield val
+
+
+class CommutationDAGNode:
+    r"""Class to store information about a quantum operation in a node of the
+    commutation DAG.
+
+    Args:
+        op (qml.Operation): PennyLane operation.
+        wires (qml.Wires): Wires on which the operation acts on.
+        node_id (int): Id of the node in the DAG.
+        successors (array[int]): List of the node's successors in the DAG.
+        predecessors (array[int]): List of the node's predecessors in the DAG.
+        reachable (bool): Attribute used to check reachability by pairewise commutation.
+        matchedwith (array[int]): Id of the matched node in the pattern.
+        isblocked (bool): Id of the matched node in the pattern.
```

Comment:
```suggestion
        matchedwith (array[int]): ID of the matched node in the pattern.
        isblocked (bool): ID of the matched node in the pattern.
```
should the description for `isblocked` be updated?